cdap-ui/app/cdap/components/HeaderActions/index.js

```diff
@@ -52,6 +88,58 @@ export default class HeaderActions extends Component {
     });
   }
 
+  toggleNamespaceDropdown(){
+    this.setState({
+      namespaceOpen : !this.state.namespaceOpen
+    });
+  }
+
+  loadNamespacesFromStore (res, current) {
+
+    if(typeof res === 'undefined'){
+      return;
+    }
+
+    if(current){
+      current = current();
+    }
+
+    this.namespaceMap = res.map( (item) => {
+
+      //If there is no current namespace, set it to default if possible
+      if(!this.state.currentNamespace && item.name === 'default'){
+        this.defaultNamespace = item.name;
+      }
+
+      let checkClass = classNames({ "fa fa-check selected-namespace-check": current === item.name });
+      let check = <span className={checkClass}></span>;
+
+      //FIXME: Currently, the <Link> does not properly include the basepath for the rendered url ; this is a problem when opening link in new tab
+      return (
+        <div key={shortid.generate()}>
+          <Link to={`/ns/${item.name}`}>
+            {check}
+            {item.name}
+          </Link>
+        </div>
+      );
+    });
+  }
+
+  selectNamespaceAndCloseDropdown(name){
```

Comment:
I think `selectNamespace` here would be sufficient . Closing the dropdown is something that I think is implicit. Doesn't need to be in the function name